ui/lib/core/addon/components/replication-action-demote.js

```diff
@@ -1,6 +1,18 @@
 import Actions from 'core/components/replication-actions-single';
 import layout from '../templates/components/replication-action-demote';
+import keys from 'vault/lib/keycodes';
 
 export default Actions.extend({
   layout,
+  onSubmit() {},
```

Comment:
You'll see this added to each replicationAction component.  The sole purpose of this is to handle the escape action.  The only two modals that don't have this are the promote and disable models, as they require you type in a confirmationText.  If I try and handle the keypress value at this level, I don't know if they've confirmed the text or not, and thus can't handle it this high up.  Instead, the escape action is handled on the ConfrimationModal level for these two components.